### PR TITLE
Added DOCS for the patchable configs in embedded chef-server in Automate.

### PR DESCRIPTION
**Pull request by talktovikas**
### :nut_and_bolt: Description: What code changed, and why?
We have an Embedded chef-server in our automate space and the configurations of chef-server can be patched by chef-automate config patch command. Recently We have more configurations which can be patched and This PR is all about telling how A user can patch these new configuration in the automate space. 

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-5900 

### :+1: Definition of Done
All the needful configs are now documentated in the docs which can be patched on Embedded chef Server by using chef-automate config patch command. 

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by lbakerchef — APPROVED** [review]
Looks good.  There are a couple failures on the verify pipeline, but it could be a broken pipeline and unrelated to your changes.

**Comment by YashviJain01** [comment]
For me, when i am seeing in chrome browers with 100% zoom(default one) aligment is not correct. 
image

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 1]
```diff
@@ -48,9 +54,13 @@ The list of parameters are as follows:
...
+| opscode_erchef['s3_external_url']             |                                      |                                            | ```[erchef.v1.sys.api]```<br>```s3_external_url="url"```         |
+| opscode_erchef['strict_search_result_acls']     |FALSE                               | FALSE                                      |``` [erchef.V1.sys.api]```<br>```strict_search_result_acls= false```|
+|opscode_erchef['enable_ibrowse_traces']          |FALSE                                      |FALSE                                |```[erchef.v1.sys.ibrowse]```<br> ```enable_ibrowse_traces = false``` |
+|license['nodes']                                 |                                           |                                     |```[erchef.v1.sys.authz]```<br>  ```node_license =```              |
+|opscode_erchef['s3_url_expiry_window_size']      | 100 Percent                               |                                     |```[erchef.v1.sys.api]``` <br> ```s3_url_expiry_window_size_percent=55```|
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:63]: this should be 100, not 100 percent?

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 2]
```diff
@@ -25,6 +24,13 @@ The list of parameters are as follows:
...
+| nginx['keepalive_timeout']                      |65                                         |65                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```keepalive_timeout=65```      |
+| nginx['sendfile']                               |ON                                         |ON                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```sendfile ="on"```            |
+| nginx['ssl_ciphers']                            |                                           |ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:AES256-GCM-SHA384:AES:!aNULL:!eNULL:!EXPORT                                     |```[cs_nginz.v1.sys.ngx.http]```<br>```ssl_ciphers=""```            |
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:33]: do we need to mention this here. @kalroy , do we have any other option.

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md (hunk 1 above)
Comment (talktovikas) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:63]: Hey @YashviJain01 , The rendered sys.config reflects { 100, percent } as the parameter. But the User Have to patch config s3_url_expiry_window_size_percent=100 in order to reflect the rendered value.

Comment (YashviJain01) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:63]: Yes, make it too 100 then. Percent is something already there in config. We have followed this while creating this doc.

Comment (talktovikas) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:63]: Yeah, Let me fix this one.

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 3]
```diff
@@ -25,6 +24,13 @@ The list of parameters are as follows:
...
+| nginx['gzip']                                   | ON                                        |ON                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip= "on"```              |
+| nginx['gzip_comp_level']                         |2                                          |2                                    |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_comp_level=2```       |
+| nginx['gzip_http_version']                       |1                                          |1                                    |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_http_version=1```     |
+| nginx['gzip_types']                              |[ 'text/plain','text/css','application/x-javascript','text/xml', 'application/xml','application/xml+rss','text/javascript','application/json']|[ 'text/plain','text/css','application/x-javascript','text/xml', 'application/xml','application/xml+rss','text/javascript','application/json']|```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_types ="[ 'text/plain','text/css','application/x-javascript','text/xml', 'application/xml','application/xml+rss','text/javascript','application/json']"```            |
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:30]:
```suggestion
| nginx['gzip_types']                              | Refer  [nginx[gzip_type]](https://docs.chef.io/server/config_rb_server_optional_settings/#nginx-14) | Refer  [nginx[gzip_type]](https://docs.chef.io/server/config_rb_server_optional_settings/#nginx-14)  |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_types ="[ 'text/plain]"```            |
```

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md (hunk 2 above)
Comment (kalroy) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:33]: We should shorten it

```suggestion
| nginx['ssl_ciphers']                            | Refer [nginx['ssl_cipgers'](https://docs.chef.io/server/config_rb_server/#nginx-ssl-protocols-14)                                          |  Refer [nginx['ssl_cipgers'](https://docs.chef.io/server/config_rb_server/#nginx-ssl-protocols-14)                              |```[cs_nginz.v1.sys.ngx.http]```<br>```ssl_ciphers=""```            |
```

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 4]
```diff
@@ -66,6 +76,9 @@ The list of parameters are as follows:
 | bookshelf['stream_download']                    | TRUE                                      | TRUE                                | ```[bookshelf.v1.sys.bookshelf]```<br>```stream_download=true```      |
 | bookshelf['aws_access_id']                    |                                       |                                 | ```[bookshelf.v1.sys.bookshelf]```<br>```aws_access_id=""```      |
 | bookshelf['aws_secret_key']                    |                                       |                                 | ```[bookshelf.v1.sys.bookshelf]```<br>```aws_secret_key=""```      |
+| bookshelf['log_rotation']                        |{ 'file_maxbytes' => 104857600, 'num_to_keep' => 10 }|                  | ```[bookshelf.v1.sys.log]```<br> ```rotation_max_bytes =104857600```<br>```rotation_max_files =10```     |
```
[the remark this answers is not in the document]
Comment (kalroy) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:79]:
```suggestion
| bookshelf['log_rotation']                        | 'file_maxbytes':  104857600, 'num_to_keep': 10 |  'file_maxbytes':  104857600, 'num_to_keep': 10 | ```[bookshelf.v1.sys.log]```<br> ```rotation_max_bytes =104857600```<br>```rotation_max_files =10```     |
```

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by dishanktiwari2501 — APPROVED** [review]

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 5]
```diff
@@ -25,6 +24,13 @@ The list of parameters are as follows:
...
 | nginx['worker_processes']                       | 4                                         | 2                                   | ```[cs_nginx.v1.sys.ngx.main]```<br>```worker_processes=4```     |
+| nginx['gzip']                                   | ON                                        |ON                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip= "on"```              |
+| nginx['gzip_comp_level']                         |2                                          |2                                    |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_comp_level=2```       |
+| nginx['gzip_http_version']                       |1                                          |1                                    |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_http_version=1```     |
+| nginx['gzip_types']                              | Refer  [nginx[gzip_type]](https://docs.chef.io/server/config_rb_server_optional_settings/#nginx-14) | Refer  [nginx[gzip_type]](https://docs.chef.io/server/config_rb_server_optional_settings/#nginx-14)  |```[cs_nginz.v1.sys.ngx.http]```<br>```gzip_types ="[ 'text/plain]"```            |
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:30]: missing single quote at the end for `text/plain`

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 6]
```diff
@@ -25,6 +24,13 @@ The list of parameters are as follows:
...
+| nginx['keepalive_timeout']                      |65                                         |65                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```keepalive_timeout=65```      |
+| nginx['sendfile']                               |ON                                         |ON                                   |```[cs_nginz.v1.sys.ngx.http]```<br>```sendfile ="on"```            |
+| nginx['ssl_ciphers']                            | Refer [nginx[ssl_cipgers]](https://docs.chef.io/server/config_rb_server/#nginx-ssl-protocols-14)                                          |  Refer [nginx[ssl_cipgers]](https://docs.chef.io/server/config_rb_server/#nginx-ssl-protocols-14)                              |```[cs_nginz.v1.sys.ngx.http]```<br>```ssl_ciphers=""```            |
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:33]: typo - ssl_cipgers for both the links

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md (hunk 5 above)
Comment (talktovikas) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md]: Hey @Dmaddu, Thank you for pointing out these, Fixing this right away.

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md (hunk 6 above)
Comment (talktovikas) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md]: Fixed

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md [hunk 7]
```diff
@@ -48,9 +54,13 @@ The list of parameters are as follows:
...
+| opscode_erchef['s3_bucket_name']             | Bookshelf                                     | Bookshelf                          | ```[erchef.v1.sys.api]```<br>```s3_bucket_name="name"```         |
+| opscode_erchef['s3_external_url']             |                                      |                                            | ```[erchef.v1.sys.api]```<br>```s3_external_url="url"```         |
+| opscode_erchef['strict_search_result_acls']     |FALSE                               | FALSE                                      |``` [erchef.V1.sys.api]```<br>```strict_search_result_acls= false```|
+|opscode_erchef['enable_ibrowse_traces']          |FALSE                                      |FALSE                                |```[erchef.v1.sys.ibrowse]```<br> ```enable_ibrowse_traces = false``` |
+|license['nodes']                                 |                                           |                                     |```[erchef.v1.sys.authz]```<br>  ```node_license =```              |
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:62]: Let's remove this line

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md (hunk 7 above)
Comment (talktovikas) [reply components/docs-chef-io/content/automate/chef_infra_in_chef_automate.md:62]: license['nodes'] This one Right ?

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**